### ISSUE TITLE
Add a QuickSave feature

### ISSUE THREAD

**Issue by fmatthew5876** [EasyRPG/Player#2046]
Most modern games support some variant of quick save nowadays.

This should be straightforward to implement.

-[ ] check for quick save key press in `Scene_Map` the same place we call debug scene.
-[ ] call `Scene_Save` which a flag which allows save only to a `QuickSave.lsd` and returns to title.
-[ ] Modify `Scene_Load` to allow loading from `QuickSave.lsd` if it is present and then delete it.

**Comment by Ghabry** [comment]
For this I would prefer a design discussion beforehand.
Also the saving requires a refactor imo. Currently parts of the save/load logic are in Scene_Load/_Save. I dislike this. Could use some kind of "Game_SaveManager". Also needed for ManiacPatch which supports this QuickLoad/-Save feature through an event command. (this also means there should be a "Schedule Save/-Load next time the interpreter returns" feature otherwise you corrupt data).

tl;dr: When designing this the ManiacPatch case should be also supported.

**Comment by fmatthew5876** [comment]
> Also the saving requires a refactor imo. Currently parts of the save/load logic are in Scene_Load/_Save. I dislike this. Could use some kind of "Game_SaveManager". 

#2015 is an attempt to solve this part of the problem

**Comment by fdelapena** [comment]
Related: #1341 and #1410.

**Comment by Br4ssman** [comment]
I'd like to know more about this.
I'm preparing a project in which I need to avoid the RM default save/load screen.

@elsemieni is helping me with a custom feature, to call save game, load game and save game information (just like Maniacs Patch offers, but instead of commands via comments xD), surely he can contribute a lot about it.

The point is that I'd need these features both for debug and for the final product. As there is no editor, I don't know how it could be used, an alternative would be to make that part of Maniac's Patch compatible in order to be able to use it through the editor.

It is a nice feature to have the possibility to make customised save and load screens.

**Comment by CortexReaver** [comment]
Is quicksave\load not implemented yet?

**Comment by Mimigris** [comment]
> Is quicksave\load not implemented yet?

There is no quicksave/load implemented yet, the closest thing you could do would be to play in Debug Mode and open the Debug Menu or press F11 to save, then load your save from the Debug Menu.
A quicksave/load feature would need to be able to be done by just pressing keys already mapped, and would need to save the content of the current battle/menu opened, hence why it is not implemented yet since currently this data cannot be saved.

**Comment by Br4ssman** [comment]
This idea would generate a temporary save that works (appears when player try to load the game) while the game is open, but disappears when the game is closed? Or would it be something else?
